crates/server/build.rs

```diff
@@ -33,63 +16,23 @@ fn main() {
         .build()
         .expect("Failed to create cache");
 
-    let options = if force {
-        Options::default().extract()
-    } else {
-        Options::default()
-    };
+    let mut options = Options::default();
+    options.extract = true;
 
-    let archive_path = cache
+    let extracted_dir = cache
         .cached_path_with_options(src, &options)
         .expect("Failed to fetch or cache UI archive");
 
-    let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap()).join("web-ui");
-    if out_dir.exists() {
-        fs::remove_dir_all(&out_dir).expect("Failed to remove existing output directory");
-    }
-    fs::create_dir_all(&out_dir).expect("Failed to create output directory");
-
-    let zip_file = fs::File::open(&archive_path).expect("Cannot open downloaded ZIP archive");
-    let mut zip = ZipArchive::new(zip_file).expect("Failed to read ZIP archive");
-    zip.extract(&out_dir)
-        .expect("Failed to extract ZIP archive");
-
-    let extracted_build_path = out_dir.join("admin-dashboard-master").join("build");
-
-    if static_files_target.exists() {
-        fs::remove_dir_all(&static_files_target).expect("Failed to clear old static files");
-    }
-    fs::create_dir_all(&static_files_target).expect("Failed to create target static files folder");
-
-    fn copy_dir_all(src: &Path, dst: &Path) {
-        for entry in fs::read_dir(src).unwrap() {
-            let entry = entry.unwrap();
-            let file_type = entry.file_type().unwrap();
-            let dst_path = dst.join(entry.file_name());
-            if file_type.is_dir() {
-                fs::create_dir_all(&dst_path).unwrap();
-                copy_dir_all(&entry.path(), &dst_path);
-            } else {
-                let _ = fs::copy(entry.path(), dst_path).expect("Failed to copy file");
-            }
-        }
-    }
-
-    copy_dir_all(&extracted_build_path, &static_files_target);
+    let extracted_build_path = PathBuf::from(&extracted_dir)
+        .join("admin-dashboard-master")
+        .join("build");
 
-    println!("cargo:rerun-if-env-changed=CALIMERO_WEB_UI_SRC");
-    println!("cargo:rerun-if-env-changed=CALIMERO_WEB_UI_FETCH");
+    println!(
+        "cargo:rustc-env=CALIMERO_WEB_UI_PATH={}",
+        extracted_build_path.display()
+    );
 
     if Path::new(src).exists() {
         println!("cargo:rerun-if-changed={}", src);
     }
```

Comment:
I don't understand this line, checking if the URL exists as a file?

shouldn't you unconditionally issue a `rerun-if-changed` on `extracted_build_path`?

Reply:
When this line is removed using `cargo run -p merod -- --node-name node1 run` will only setup this once. If you manually delete the target folder it won't re-run it unless the build.rs file is updated again. And adding it will run the build.rs when you start the node and not have target folder already. 

Just covering this case..

Reply:
what confuses me is the `Path::new(src)`

when the path is a URL, won't that be false?

and also

1. at this point in the code, isn't it already guaranteed we have the web UI path existing?
2. we should unconditionally rerun the build script if the dir changes